tools/toolchains/arm.py

```diff
@@ -49,7 +49,7 @@ def __init__(self, target, notify=None, macros=None,
                                extra_verbose=extra_verbose,
                                build_profile=build_profile)
 
-        if "ARM" not in target.supported_toolchains:
+        if "ARM" not in target.supported_toolchains and "uARM" not in target.supported_toolchains:
```

Comment:
This is not a workable solution as it allows compiles of `uARM` only targets with `ARM`.